cudamapper/include/claraparabricks/genomeworks/cudamapper/overlapper.hpp

```diff
@@ -72,12 +73,16 @@ class Overlapper
     /// \brief returns overlaps for a set of reads
     /// \param fused_overlaps Output vector into which generated overlaps will be placed
     /// \param d_anchors vector of anchors sorted by query_read_id -> target_read_id -> query_position_in_read -> target_position_in_read (meaning sorted by query_read_id, then within a group of anchors with the same value of query_read_id sorted by target_read_id and so on)
+    /// \param query_index A shared pointer to an index of sketch elements for the queries.
+    /// \param target_index A shared pointer to an index of sketch elements for the targets.
     /// \param min_residues smallest number of residues (anchors) for an overlap to be accepted
     /// \param min_overlap_len the smallest overlap distance which is accepted
     /// \param min_bases_per_residue the minimum number of nucleotides per residue (e.g minimizer) in an overlap
     /// \param min_overlap_fraction the minimum ratio between the shortest and longest of the target and query components of an overlap. e.g if Query range is (150,1000) and target range is (1000,2000) then overlap fraction is 0.85
     virtual void get_overlaps(std::vector<Overlap>& fused_overlaps,
                               const device_buffer<Anchor>& d_anchors,
+                              std::shared_ptr<Index> query_index,
```

Comment:
`ApplicationParameters` has `all_to_all` member (https://github.com/clara-parabricks/GenomeWorks/blob/dev-v0.5.0/cudamapper/src/application_parameters.hpp#L62). I think passing that bool is cleaner that passing both indices, plus that way we have a centralized place where we decide if we are doing map to reference or all to all.

Reply:
@vellamike and I debated this a bit, since `all_to_all` gets passed around less frequently than indices do. I think you're right though, and this prevents the awkward pointer comparison I do in its place.

Reply:
OK. Thinking about it more and accounting for the failing test and I agree, let's go with passing in `all_to_all`